### PR TITLE
Pt34 login

### PR DESCRIPTION
**Pull request by Felipe-Cadavid**
Login backend now responds with a JWT on login

### CODE REVIEW

File: middlewares/logger.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const Student = require('../models/student.model');
+const Tutor = require('../models/tutor.model');
+const jwt = require('jsonwebtoken');
+
+const logger = async (req, res, next) => {
+    try {
+        const { email, password } = req.body;
```
Comment (leoneloliveros) [inline middlewares/logger.js:7]: This two conditions are almost the same, you can use one try - catch and add conditions in try
calling their respective model, something like this:

image

File: models/student.model.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+const bcrypt = require('bcrypt')
+
+const studentSchema = mongoose.Schema({
+  email: {
+    type: String,
+    match: /.+\@.+\..+/,
+    required: [true, "El email es requerido"],
+    validate: {
+      validator: async function(value) {
+        const student = await Student.findOne({ email: value })
```
Comment (leoneloliveros) [inline models/student.model.js:11]: What happen is Student and tutor have the same email, can they be duplicated?

File: models/tutor.model.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+const bcrypt = require('bcrypt')
+
+const tutorSchema = mongoose.Schema({
+  email: {
+    type: String,
+    match: /.+\@.+\..+/,
+    required: [true, "El email es requerido"],
+    validate: {
+      validator: async function(value) {
+        const tutor = await Tutor.findOne({ email: value })
```
Comment (leoneloliveros) [inline models/tutor.model.js:11]: Same here! can they be duplicated?

File: middlewares/logger.js (hunk 1 above)
Comment (Felipe-Cadavid) [reply middlewares/logger.js:7]: Hello Leonel, thank you for your review, I made a refactor on the logger

File: models/student.model.js (hunk 2 above)
Comment (Felipe-Cadavid) [reply models/student.model.js]: There can not be a student and a tutor with the same email, I added an extra validation

File: models/tutor.model.js (hunk 3 above)
Comment (Felipe-Cadavid) [reply models/tutor.model.js]: Yes, I just added an extra validation

File: middlewares/logger.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        let user;
+        const student = await Student.authenticate(email, password);
+        if (student) {
```
Comment (leoneloliveros) [inline middlewares/logger.js:10]: You can refactor a little more

Comment (Felipe-Cadavid) [reply middlewares/logger.js]: Hello Leonel, I've done another refactor, I think now it's much cleaner

**Review by leoneloliveros — APPROVED** [review]
Great! That's it

**Pull request merged**